src/components/checkbox/checkbox.js

```diff
@@ -171,12 +197,48 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
+        if (indeterminateStateEnabled) {
+          setIndeterminateState();
+          // The checkbox should never be in an indeterminate="true" AND checked state.
+          // To prevent this, if the checkbox is in an indetermiante
+          // state, we skip the code below.
+          return;
+        }
+
         if(ngModelCtrl.$viewValue) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState() {
+        var isIndeterminate = element.attr('indeterminate') === 'true' || false;
```

Comment:
@topherfangio I agree with you. That's exactly what `$mdUtil.parseAttributeValue` does.

But notice, attributes like `readonly`, `disabled` or `required` should ignore false values.
That's why you should use in that case
```js
$mdUtil.parseAttributeValue(attr, false)
```